### ISSUE TITLE
Generating a slug from a title afterStateUpdated $set()

### ISSUE THREAD

**Issue by csandreas1** [laravel-filament/filament#12326]
### Package

filament/filament

### Package Version

3

### Laravel Version

11

### Livewire Version

3

### PHP Version

8.3.0

### Problem description

Followed this guide:
https://filamentphp.com/docs/3.x/forms/advanced#generating-a-slug-from-a-title 

There seems to be a bug, see video.
I attempt to write Hello World but some text is automatically deleted.
[Screencast from 2024-04-14 16-51-13.webm](https://github.com/filamentphp/filament/assets/21238024/73a6cb50-ffc9-4450-8591-41a05c548efa)

Note: When i type slowly it works

### Expected behavior

Slug field should show "Hello World"

### Steps to reproduce

follow the tutorial.

### Reproduction repository

https://github.com/csandreas1/event-system

**Comment by jcc5018** [comment]
I am having the same issue after copying the exact code in the documentation.   The title fields keep removing what I am typing in as it is sending an update after every character and not keeping up with the slug generation.  

could this functionality be improved with alpine?  I'm not sure what i would need to edit to make this happen, but this shouldnt need to send request when basic JS functions accomplish the same goal. Right?

**Comment by faridsa** [comment]
I'm facing the same situation. When typed slowly in title field, slug field is updated accurately, but when typed at normal speed cursor does backspace removing some characters.

It could be ideal to get slug field filled only when title field is blurred, because it does mean that title is completely writen

**EDIT:**
Just for future reference, it could be solve using `->live(onBlur: true)` instead of just `->live()`

**Comment by neil-cldagency** [comment]
Hi there — apologies for resurrecting this but I've just had the exact same issue after following the same guidance on a fresh resource. The slug field will work well but if you type quickly, the letters in the originating (title) field seem to be getting wiped — as if the process is not quite up to date with the field's current value and is lagging behind.

**Comment by neil-cldagency** [comment]
I've tried multiple methods — adding a `debounce` value works but it falls apart eventually no matter how high the number is. `onBlur` argument to live seems to be the most robust but not as good UX as it updating live.

In some cases I did notice intermediate XHRs to `livewire/update` that contain an out of sync value for the field being monitored.

I think it may be something to do with `$set`? When it's not being used, a log of the current `$state` within the `afterStateUpdated` callback seems to be accurate every time.
